src/bytes.js

```diff
@@ -0,0 +1,1443 @@
+            arr = new Uint8Array(source);
+        } else if (source instanceof Sk.builtin.bytes) {
+            return source;
+        } else if (Array.isArray(source)) {
+            // Internal fast path
+            Sk.asserts.assert(source.every((x) => (x >= 0) && (x < 256)),
+                              "Bad internal call to bytes with array object");
+            arr = new Uint8Array(source);
+        } else if (source instanceof Uint8Array) {
+            // Internal fast path
+            arr = source;
+        } else if ((Sk.builtin.checkIterable(source) && !(source instanceof Sk.builtin.str))) {
+            const final = [];
+            for (let iter = Sk.abstr.iter(source), item = iter.tp$iternext();
+                item !== undefined;
+                item = iter.tp$iternext()) {
+                if (Sk.builtin.checkInt(item)) {
+                    item = Sk.builtin.asnum$(item);
+                    if (item >= 0 && item <= 256) {
+                        final.push(item);
+                    } else {
+                        throw new Sk.builtin.ValueError("bytes must be in range(0, 256)");
+                    }
+                } else {
+                    throw new Sk.builtin.TypeError("'" + Sk.abstr.typeName(item) + "' " + "object cannot be interpreted as an integer");
+                }
+            }
+            arr = new Uint8Array(final);
+        } else if ((source instanceof Sk.builtin.str) || (typeof source === "string")) {
+            throw new Sk.builtin.TypeError("string argument without an encoding");
+        } else {
+            throw new Sk.builtin.TypeError("cannot convert '" + Sk.abstr.typeName(source) + "' object into bytes");
+        }
+    } else if (arguments.length > 1) {
+        if (encoding instanceof Sk.builtin.str) {
+            if ((source instanceof Sk.builtin.str) || (typeof source === "string")){
+                if (source instanceof Sk.builtin.str) {
+                    source = source.$jsstr();
+                }
+
+                encoding = normalizeEncoding(encoding.$jsstr());
+                if (encoding === "ascii") {
+                    const data = [];
+                    for (let i in source) {
+                        const val = source[i].charCodeAt(0);
+
+                        if (val < 0 || val > 127) {
+                            if (errors === "strict") {
+                                const hexval = makehexform(val);
+                                throw new Sk.builtin.UnicodeEncodeError("'ascii' codec can't encode character '" + hexval + "' in position " + i + ": ordinal not in range(128)");
+                            } else if (errors === "replace") {
+                                data.push(63); // "?"
+                            }
+                        } else {
+                            data.push(val);
+                        }
+                    }
+                    arr = Uint8Array.from(data);
```

Comment:
same question 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/from#Browser_compatibility

Reply:
I wrote a really simple polyfill for Uint8Array.from and I use "from" everywhere an array is converted to a Uint8Array now.  The polyfill is good enough for the uses in the file, but I'm not sure if people are comfortable with that or not.   I don't know if we want a full Uint8Array polyfill, though a cursory look didn't turn up one that I was happy with.

Reply:
The more I think about it, the less okay I am with a sketchy polyfill. I'll look harder for a better one, and if there isn't one, then perhaps just write a function that does what we want with a clear comment about why it doesn't use the from method.

Reply:
The more I think about it, the less okay I am with a sketchy polyfill. I'll look harder for a better one, and if there isn't one, then perhaps just write a function that does what we want with a clear comment about why it doesn't use the from method.

Reply:
The more I think about it, the less okay I am with a sketchy polyfill. I'll look harder for a better one, and if there isn't one, then perhaps just write a function that does what we want with a clear comment about why it doesn't use the from method.

Reply:
The more I think about it, the less okay I am with a sketchy polyfill. I'll look harder for a better one, and if there isn't one, then perhaps just write a function that does what we want with a clear comment about why it doesn't use the from method.

Reply:
The more I think about it, the less okay I am with a sketchy polyfill. I'll look harder for a better one, and if there isn't one, then perhaps just write a function that does what we want with a clear comment about why it doesn't use the from method.

Reply:
The more I think about it, the less okay I am with a sketchy polyfill. I'll look harder for a better one, and if there isn't one, then perhaps just write a function that does what we want with a clear comment about why it doesn't use the from method.